source/io/CsvWriter.ooc

```diff
@@ -0,0 +1,31 @@
+use ooc-base
+use ooc-collections
+use ooc-io
+import io/[File, FileWriter]
+
+CsvWriter: class {
+	_fileWriter: FileWriter
+	init: func (=_fileWriter)
+	free: func {
+		if (this _fileWriter != null) {
+			this _fileWriter close()
+			this _fileWriter free()
+		}
+		super()
+	}
+	write: func (row: VectorList<Text>) {
+		for (i in 0 .. row count) {
+			this _fileWriter file write(row[i] toString())
+			if (i < row count - 1)
+				this _fileWriter write(";")
+		}
+		this _fileWriter write("\n")
+	}
+	open: static func (filename: Text) -> This {
+		result: This
+		file := File new(filename toString())
+		result = This new(FileWriter new(file))
+		file free()
```

Comment:
The whole function could be written as 
```ooc
This new(FileWriter new(filename toString()))
```
unless that causes a `String` leak, of course.